elyra/pipeline/parser.py

```diff
@@ -234,3 +242,42 @@ def _scrub_list(dirty: Optional[List[Optional[str]]]) -> List[str]:
         if not dirty:
             return []
         return [clean for clean in dirty if clean]
+
+    def _get_remaining_component_params(self, node: Dict):
+        """
+        Builds a dictionary of the parameters for a given node that do not have a corresponding
+        property in the Operation object. These parameters will be used by the appropriate processor when
+        loading and running a component that is not one of the standard notebook or script operations.
+        """
+        component_params = {}
+        if node.get('op') not in ["execute-notebook-node", "execute-python-node", "execute-r-node"]:
+            for key, value in node['app_data'].items():
+                # Do not include any null values
+                if not value:
+                    continue
+                # Do not include any of the standard set of parameters
+                if key in ["filename", "runtime_image", "cpu", "gpu", "memory", "dependencies", "env_vars", "outputs",
+                           "include_subdirectories", "ui_data", "component_source", "component_source_type",
+                           "elyra_airflow_class_name"]:
+                    continue
+                # For Airflow inputs, remove class name information from key
+                class_name = PipelineParser._get_app_data_field(node, 'elyra_airflow_class_name')
+                if class_name and not key.startswith(class_name.lower().replace(' ', '_')):
```

Comment:
Per the previous comment regarding `elyra_airflow_class_name` being an empty list, I guess I'm still confused why its an empty list here.  Neither of the `class_name.lower()` calls will work.  Here's a screenshot from the debugger.
![image](https://user-images.githubusercontent.com/22599560/121089992-c6db0480-c79c-11eb-86bc-a41618071485.png)
If we're using the empty list as a `None` indicator, we should use the latter since `class_name` should ultimately be a string.

Reply:
I think it's only an empty list here because a class wasn't selected from the dropdown menu. I'm testing it out now, but ideally what will happen when I make `elyra_airflow_class_name` a required field is that it will force the user to select one of the options from the dropdown. And as such, only the selected class will be passed as a string to the backend